src/modules/dashboard/components/table-action-buttons.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div class="action-buttons">
+    <v-btn
+      icon="mdi-pencil"
+      size="x-small"
+      @click="emit('edit', itemId)"
+    />
+    <v-btn
+      icon="mdi-delete"
+      size="x-small"
+      @click="emit('delete', itemId)"
+    />
+  </div>
+</template>
+
+<script setup lang="ts" generic="T">
+const emit = defineEmits<{
+  (e: 'edit', itemId: T): void;
+  (e: 'delete', itemId: T): void;
+}>();
```

Comment:
Tak jak poprzednio (chyba że nie czujesz tego podejścia), typy najlepiej wynieść poza definEmits + może typy akcji przenieść do enuma? nie będziemy wtedy operować stringami

Reply:
Swoją drogą itemId to nie będzie number/string? Musi tu być generyk?